routes/dao_coin_exchange_with_fees.go

```diff
@@ -0,0 +1,1602 @@
+		0,                             // Don't add additionalFees
+		additionalExtraData,           // The new ExtraData
+		requestData.MinFeeRateNanosPerKB, fes.backendServer.GetMempool(), []*lib.DeSoOutput{})
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("UpdateDaoCoinMarketFees: Problem creating transaction: %v", err))
+		return
+	}
+
+	txnBytes, err := txn.ToBytes(true)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("UpdateDaoCoinMarketFees: Problem serializing transaction: %v", err))
+		return
+	}
+
+	// Return all the data associated with the transaction in the response
+	res := UpdateDaoCoinMarketFeesResponse{
+		TotalInputNanos:   totalInput,
+		ChangeAmountNanos: changeAmount,
+		FeeNanos:          fees,
+		Transaction:       txn,
+		TransactionHex:    hex.EncodeToString(txnBytes),
+		TxnHashHex:        txn.Hash().String(),
+	}
+	if err = json.NewEncoder(ww).Encode(res); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("UpdateDaoCoinMarketFees: Problem encoding response as JSON: %v", err))
+		return
+	}
+}
+
+type GetDaoCoinMarketFeesRequest struct {
+	ProfilePublicKeyBase58Check string `safeForLogging:"true"`
+}
+
+type GetDaoCoinMarketFeesResponse struct {
+	FeeBasisPointsByPublicKey map[string]uint64 `safeForLogging:"true"`
+}
+
+func GetTradingFeesForMarket(
+	utxoView *lib.UtxoView,
+	params *lib.DeSoParams,
+	profilePublicKey string,
+) (
+	_feeMapByPubkey map[string]uint64,
+	_err error,
+) {
+
+	// Decode the public key
+	profilePublicKeyBytes, _, err := lib.Base58CheckDecode(profilePublicKey)
+	if err != nil || len(profilePublicKeyBytes) != btcec.PubKeyBytesLenCompressed {
+		return nil, fmt.Errorf(
+			"GetTradingFeesForMarket: Problem decoding public key %s: %v",
+			profilePublicKey, err)
+	}
+
+	// Pull the existing profile. If one doesn't exist, then we error. The user should
+	// create a profile first before trying to update the fee params for their market.
+	existingProfileEntry := utxoView.GetProfileEntryForPublicKey(profilePublicKeyBytes)
+	if existingProfileEntry == nil || existingProfileEntry.IsDeleted() {
+		return nil, fmt.Errorf(
+			"GetTradingFeesForMarket: Profile for public key %v does not exist",
+			profilePublicKey)
+	}
+
+	// Decode the trading fees from the profile.
+	tradingFeesByPkidBytes, exists := existingProfileEntry.ExtraData[lib.TokenTradingFeesByPkidMapKey]
+	tradingFeesMapPubkey := make(map[lib.PublicKey]uint64)
+	if exists {
+		tradingFeesMapByPkid, err := lib.DeserializePubKeyToUint64Map(tradingFeesByPkidBytes)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"GetTradingFeesForMarket: Problem deserializing trading fees: %v", err)
+		}
+		for pkid, feeBasisPoints := range tradingFeesMapByPkid {
+			pkidBytes := pkid.ToBytes()
+			if len(pkidBytes) != btcec.PubKeyBytesLenCompressed {
+				return nil, fmt.Errorf(
+					"GetTradingFeesForMarket: Problem decoding public key %s: %v",
+					pkid, err)
+			}
+			pubkey := lib.PKIDToPublicKey(lib.NewPKID(pkidBytes))
```

Comment:
This is incorrect - we want to use a utxo view to fetch the public key for the given PKID instead of just converting the PKID to a public key. If a public key had its identity swapped, then this would behave unexpectedly.

Reply:
Whoops! Good catch I fixed it. Don't think there are any other places where this was an issue.